### DISCUSSION TITLE
Top Languages card not showing any data

### DISCUSSION THREAD

**Discussion by NickB-30** [anuraghazra/github-readme-stats#1878] [Q&A] [answered]
**Describe the bug**
The top languages card does not show anything.

**Expected behaviour**
The card displays my most used languages

**Screenshots / Live demo link (paste the github-readme-stats link as markdown image)**
<img width="611" alt="image" src="https://user-images.githubusercontent.com/80864505/180282419-f2915d4e-ad90-4e51-8a81-0f0e1e5f944a.png">

**Additional context**
I have 2 private repos which use different languages

**Comment by rickstaa** [comment, marked as answer]
@NickB-30 Thanks for your question. I think your issue is related to #1789 and [FAQ#language-card-related-questions](https://github.com/anuraghazra/github-readme-stats/discussions/1770#language-card-related-questions). I will, however, also answer it here. I checked your username's stats card and believe the results are correct as you don't have any repositories from which GitHub can report language results. You can check this out yourself by using the following query in the [graphql explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

You can then use the following in the query variables field.

```graphql
{"login": "NickB-30"}
```

The public Vercel instance cannot access your private repositories, so it can not show language results. If you want the result for your private repositories to show up you can create your own vercel instance (see https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance).

**Reply by assebc** [reply]
@rickstaa Today is happening exactly the same to me, yesterday when I first deploy the repository it was working perfectly but today something happened and it just don't load, can you help me figured it out?

**Reply by rickstaa** [reply]
Hey @assebc. No problem. If you followed [this guide](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) your private contributions should show up if you correctly added the `PAT_1` token to the Vercel instance and are using the right Vercel deployment URL. The URL is visible under the `DOMAINS` of Vercels overview tab. Replace the public Vercel URL (i.e. https://github-readme-stats.vercel.app) with this URL, and you should be good to go.

**Reply by assebc** [reply]
Thank you @rickstaa for your help, but even though I followed the guide, have the PAT_1 added correctly and using my vercel domain I am able to see other features of github-readme-stats but I can not see my Top-Languages Card, what should I do or where can it be the problem?

**Reply by rickstaa** [reply]
@assebc To debug your problem, you could check if the GraphQL explorer gives you any data when you log in (see https://github.com/anuraghazra/github-readme-stats/discussions/1878#discussioncomment-3200472). This is precisely the data we get in the app. If you do not receive any data, the card is displaying correctly. Please be aware that linguist does not consider markdown as a language by default (see https://github.com/github/linguist/issues/5951). Further, you have to make sure that you are using the correct Vercel deployment URL (i.e. DOMAIN) and that the deployment is on the right commit (i.e. under BRANCH):

![image](https://user-images.githubusercontent.com/17570430/182571753-7a2cafde-b214-4cfd-ae46-b0155b704805.png)

**Reply by assebc** [reply]
@rickstaa I already tested it and got this answer:

![image](https://user-images.githubusercontent.com/73396142/182575080-9d12e3e9-8ba1-430a-bee6-334d4d6abc3b.png)

I don't know if is that what I want to receive or I wanted something else, remind that in (login:  $login) I put (login: "assebc"). I also checked if it is indeed in the right commit and it confirmes! I also tested my domain with other usernames and it worked, don't know why it does not function with mine. Ps: The original one works fine, but I created the deployed version to check my private repositories!

**Reply by rickstaa** [reply]
@essebc That error means that your syntax is not correct. You should place the `login` in the variable field like so:

![image](https://user-images.githubusercontent.com/17570430/182592174-7f468fa4-3b7b-4ac1-b9f4-46ccf094bea9.png)

Since I can not reproduce the issue, I am afraid I can not help you any further. You can build and debug the code on your local computer using the [Vscode debugger](https://code.visualstudio.com/docs/editor/debugging). This will allow you to check which data or errors are returned using your username (see https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md and https://github.com/vercel/vercel/issues/2864#issuecomment-963533339https://github.com/vercel/vercel/issues/2864#issuecomment-963533339 for more info).

**Reply by assebc** [reply]
Thank you so much @rickstaa you helped a lot!

**Reply by rickstaa** [reply]
I also noticed that you orphaned the repository instead of forking it (see https://github.com/assebc/github-readme-stats). In that case, I am afraid I cannot help you since the commit histories are unrelated. I would advise you not to orphan repositories but to work with forks as this aligns with the Git philosophy. The following syntax should work with your custom URL when you deploy the Vercel instance from a fork:

```md
[![Top Langs](https://<YOUR_CUSTOM_URL>/api/top-langs/?username=assebc)](https://github.com/anuraghazra/github-readme-stats)
```

[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=assebc)](https://github.com/anuraghazra/github-readme-stats)

**Reply by rickstaa** [reply]
An additional benefit of using forks instead of orphans is that you can easily keep the fork up to date with, for example, the sync fork button:

![image](https://user-images.githubusercontent.com/17570430/182593839-7b55c6b8-dd2c-4521-8fb1-5f718e8fac4e.png)

**Comment by Swapnaroop2001** [comment]
My language card is not fetching data
Please help

<img width="1280" alt="Screen Shot 2023-11-15 at 11 05 58 PM" src="https://github.com/anuraghazra/github-readme-stats/assets/75776020/729e4c78-a057-4a3d-8003-472120896877">

**Reply by qwerty541** [reply]
Hey, @Swapnaroop2001! The card shows message "No languages data," because it actually doesn't have any data to show. It doesn't count forked repositories.

**Reply by vitorabujamra** [reply]
Count private repositories?
I create a Vercel App for myself and yet dont show any data

https://github-readme-stats-lime-kappa.vercel.app/api/top-langs/?username=vitorabujamra&theme=dark
Even my username dont show my private commits

https://github-readme-stats-lime-kappa.vercel.app/api?username=vitorabujamra&include_all_commits=true&count_private=true&show_icons=true&theme=dark

**Reply by qwerty541** [reply]
@vitorabujamra To show your private repositories data on the card you have to use own Vercel deployment and GitHub personal access token with enabled `repo` and `user` permissions as on screenshots.

![image](https://github.com/anuraghazra/github-readme-stats/assets/53787217/9c94bb06-a59d-402e-8014-5f821046b630)

![image](https://github.com/anuraghazra/github-readme-stats/assets/53787217/c835b969-9850-4cc9-82ed-b9207214b47f)

**Reply by vitorabujamra** [reply]
I just did it!
It work!!

Thanks!!

**Comment by code-with-sayad-babu-dev** [comment]
Hi @NickB-30 👋

I faced the same issue, and after some research, I found a few possible reasons:

1. 🔒 The Top Languages card only works with **public repositories**. If most of your repos are private, the card will be blank.
2. 🛠️ If your repos don't contain detectable code files (like `.js`, `.php`, `.py`, etc.), GitHub Stats can't calculate language usage.
3. 🔄 Sometimes it takes time to update newly created or modified repos. Try clearing cache or visiting the link directly in browser.
4. 🧪 Also, forked repos are ignored unless you’ve contributed actual code.

Try this link in your browser directly to test it:
https://github-readme-stats.vercel.app/api/top-langs/?username=your-username&layout=compact

vbnet
Copy
Edit

Hope it helps. 😊 Let me know if you find a different solution too!

Thanks,  
Sayad Babu

**Reply by RumeshChathuranga** [reply]
Now this Deployment has been paused by the owner. Is there any other way or other repo to use instead?

<img width="788" height="517" alt="image" src="https://github.com/user-attachments/assets/2d2115a9-c513-42da-9acc-248c52966991" />

**Reply by qwerty541** [reply]
@RumeshChathuranga Hello

You can deploy your own instance using your own GitHub API token, here is the documentation https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#deploy-on-your-own
